cpu/rp2350/xosc.c

```diff
@@ -0,0 +1,56 @@
+/*
+ * SPDX-FileCopyrightText: 2025 Tom Hert <git@annsann.eu>
+ * SPDX-FileCopyrightText: 2025 HAW Hamburg
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+/**
+ * @ingroup         cpu_rp2350
+ * @{
+ *
+ * @file
+ * @brief           XOSC implementation for the RP2350
+ *
+ * @author          Tom Hert <git@annsann.eu>
+ */
+
+#include <assert.h>
+
+#include "RP2350.h"
+#include "board.h"
+#include "macros/units.h"
+#include "periph_cpu.h"
+
+/* Based on datasheet 8.2.4 (1ms wait time) */
+#define STARTUP_DELAY 47
+#define MAX_XOSC_COUNTER_SIZE 0xFFFF
+#define SLEEP_100HZ_SPEED 60000
```

Comment:
```suggestion
#define SLEEP_100HZ_SPEED 12000000UL
```

Although this shouldn't be a hard coded value I guess.